packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -186,21 +195,31 @@ export class TreeWidget extends VirtualWidget implements StatefulWidget {
         };
     }
 
-    protected renderSubTree(node: TreeNode, props: NodeProps): h.Child {
+    protected renderSubTree(node: TreeNode, props: NodeProps): React.ReactNode {
         const children = this.renderNodeChildren(node, props);
         if (!TreeNode.isVisible(node)) {
             return children;
         }
         const parent = this.renderNode(node, props);
-        return VirtualRenderer.merge(parent, children);
+        return <React.Fragment key={node.id}>{parent}{children}</React.Fragment>;
     }
 
-    protected renderIcon(node: TreeNode, props: NodeProps): h.Child {
+    protected renderIcon(node: TreeNode, props: NodeProps): React.ReactNode {
         // tslint:disable-next-line:no-null-keyword
         return null;
     }
 
-    protected renderExpansionToggle(node: TreeNode, props: NodeProps): h.Child {
+    protected readonly toggle = (event: React.MouseEvent<HTMLElement>) => this.doToggle(event);
+    protected doToggle(event: React.MouseEvent<HTMLElement>) {
+        const nodeId = event.currentTarget.getAttribute('data-node-id');
+        if (nodeId) {
+            const node = this.model.getNode(nodeId || undefined);
```

Comment:
- ` || undefined`. You already have the guard against `null`.

Reply:
Sorry, forgot that.